src/test/shape_test.cpp

```diff
@@ -50,6 +51,27 @@ TEST(Dimension, PartiallyFitsInto) {
     EXPECT_FALSE(Dimension(1, 2).partiallyFitsInto(Dimension(3, 4)));
 }
 
+TEST(Dimension, CreateIntersection) {
+    EXPECT_EQ(Dimension(1, 2).createIntersection(Dimension::any()), Dimension(1, 2));
+    EXPECT_EQ(Dimension(1, 1).createIntersection(Dimension::any()), Dimension(1, 1));
```

Comment:
We should not create static dimension using this constructor. This only works because we do not yet throw an exception during construction.